hubclient/array-chunk-iterator.go

```diff
@@ -0,0 +1,30 @@
+package hubclient
+
+type ArrayChunkIterator struct {
+	chunks   []string
+	position int
+}
+
+func NewArrayChunkIterator(chunks []string) ArrayChunkIterator {
+	i := new(ArrayChunkIterator)
+	i.chunks = chunks
+	i.position = -1
+	return *i
+}
+
+func (i *ArrayChunkIterator) hasNext() bool {
+	if len(i.chunks) > (i.position + 1) {
+		return true
+	} else {
+		return false
+	}
+}
+
+// next returns true and the next chunk if there is a next chunk. Returns false otherwise.
+func (i *ArrayChunkIterator) next() (string, bool) {
```

Comment:
Instead of returning bool, should this return err? From what I am reading it seems like this is a check that the caller first called hasNext before calling next (if they call next more than once it may error)

Reply:
Right, I wanted some protection around callers who weren't properly checking via hasNext first. I can rework to return an error instead.